### PR TITLE
Update to e2e toolchain

### PR DESCRIPTION
**Pull request by SethDavenport**
No more gulp - instead we use a shell script + `screen` to manage the various
processes.

Updated the tests to use arrow functions since it's more concise.

Added an optional 'wait' into the tests to make them easier to demo.

Fixed some `tsconfig.json` from recent lessons learned in other typescript projects.

### CODE REVIEW

**Comment by SethDavenport** [comment]
@igor-ka @bennett000

This is what I'm presenting at Rangle U - if we like this approach then I'll roll it over to the starters.

**Comment by igor-ka** [comment]
@SethDavenport LGTM, the only issue I have is that shell scripts are platform specific.
This is more of a problem for ngcourse-next as a lot of people come with Windows laptops.

This is less of a concern with the clients, although some of them do use windows.
I am wondering if this is a legit concern as at home I use git-smc and I get full Linux capable terminal with that.

Any thoughts?

File: tsconfig.json [hunk 1]
```diff
@@ -8,8 +8,11 @@
     "removeComments": false,
     "sourceMap": true
   },
+  "compileOnSave": false,
+  "buildOnSave": false,
+  "files": [ "app/__typings/tsd.d.ts" ],
   "exclude": [
     "node_modules",
     "app/__build"
```
Comment (igor-ka) [inline tsconfig.json]: is this still called __build as opposed to dist?

**Comment by SethDavenport** [comment]
It's a reasonable concern.  Choreographing the various procs is kind of a pain.  I wonder if there's a similar thing to screen on windows?

Otherwise we'd need to require the use of cygwin or similar which may not be popular.

**Comment by SethDavenport** [comment]
Unfortunately I don't have easy access to a windows machine at the moment.

**Comment by SethDavenport** [comment]
I wonder if this might help: https://github.com/shelljs/shelljs

**Comment by igor-ka** [comment]
Seth, shelljs looks promising we should try it out. 

I am wondering if ngcourse-next is the right place for this. When we will teach it again, and as I understand we will, I am not sure what to do with the windows users that will take this course. I originally done something similar and reverted it right before the class, realizing that a lot of people are on Windows.

**Comment by SethDavenport** [comment]
Yeah I'll put it in the starters instead.  I added the original implementation to ngcourse next a while ago for rangle-u convenience, but that was pre-starter.

**Pull request closed without merging**